### PR TITLE
Added lot of cases for fixtures

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: source/parser.apib [hunk 1]
```diff
@@ -32,7 +32,7 @@ application/vnd.refract.parse-result+json
 application/vnd.refract.parse-result+yaml
 ```
 
-General-purpose result of the parsing operation. The parse result is in form of the Refract data structure as defined in its [specification](https://github.com/refractproject/refract-spec). The parse result data comply with the [Parse Result Namespace](https://github.com/refractproject/refract-spec/blob/master/namespaces/parse-result.md).
+General-purpose result of the parsing operation. The parse result is in form of the Refract data structure as defined in its [specification](https://github.com/refractproject/refract-spec). The parse result data comply with the [Parse Result Namespace](https://github.com/refractproject/refract-spec/blob/master/namespaces/parse-result-namespace.md).
```
Comment (kylef) [inline source/parser.apib]: Should probably link to API Elements instead?

http://api-elements.readthedocs.io/en/latest/

File: Makefile [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
-	sample.refract.parse-result.yaml \
-	sample.apiblueprint.parse-result.json \
-	sample.apiblueprint.parse-result.yaml \
-	sample.apiblueprint.ast-2.0.yaml \
-	sample.apiblueprint.ast-2.0.json
-DEPENDENCIES = $(foreach file,$(SOURCES),source/$(file))
 HOST := https://api.apiblueprint.org/
 APIARY_API := apiblueprintapi
 
-apiary.apib: node_modules $(DEPENDENCIES)
```
Comment (kylef) [inline Makefile]: Why this change? This means if I run `make apiary.apib`, then make a change to one of the included files and run `make apiary.apib` again it won't be updated.

File: source/dredd-hooks.js [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
-  if (transaction.expected.headers['Content-Type'] === 'text/vnd.apiblueprint') {
-    transaction.real.body = transaction.expected.body.replace(/^$/g, '');
+  var type = typer.parse(transaction.expected.headers['Content-Type']);
+
+  // Indent the request and response bodies properly
+  if (type.suffix === 'json' || type.subtype === 'json') {
+    transaction.expected.body = JSON.stringify(JSON.parse(transaction.expected.body));
+    transaction.real.body = JSON.stringify(JSON.parse(transaction.real.body));
+
+    // Change content type because otherwise gavel would compare JSON objects
```
Comment (kylef) [inline source/dredd-hooks.js]: Wouldn't this mean that we would no longer validate the real content-type is correct? We only checked that it was JSON before, now we are setting it to `text/plain` without verifying that the full expected content-type matches the real content type.

This hook would miss implementation problems such as if the server sent the wrong content type such as if there was a typo or a different version.

If we're going to set to `text/plain`, please validate the content-types match beforehand.

**Comment by pksunkara** [comment]
Updated.
